src/app.hooks.js

```diff
@@ -102,27 +102,15 @@ const handleAutoLogout = async (context) => {
  */
 const errorHandler = (context) => {
 	if (context.error) {
-		context.error.code = context.error.code || context.error.statusCode;
-		if (!context.error.code && !context.error.type) {
-			const catchedError = context.error;
-			if (catchedError.hook) {
-				// too much for logging...
-				delete catchedError.hook;
-			}
-			context.error = new GeneralError(context.error.message || 'Server Error', context.error.stack);
-			context.error.catchedError = catchedError;
-		}
-		context.error.code = context.error.code || 500;
+		context.error.code = context.error.code || context.error.statusCode || 500;
 
 		if (context.error.hook) {
 			// too much for logging...
 			delete context.error.hook;
 		}
 		return context;
 	}
-	context.app.logger.warning('Error with no error key is throw. Error logic can not handle it.');
-
-	throw new GeneralError('server error');
+	throw new GeneralError('Server error!', prepareErrorParam('Error with no context.error is throw. Error logic can not handle it.'));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This line has a length of 135. Maximum allowed is 120.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5258109)